include/executors/executor_sycl.hpp

```diff
@@ -287,124 +286,116 @@ class Executor<SYCL> {
-      auto acc =
-          blas::get_range_accessor<cl::sycl::access::mode::write>(dst, cgh);
-      cgh.copy(src, acc);
-    });
+  template <typename ContainerT0, typename ContainerT1>
+  inline auto copy_to_host(ContainerT0 src, ContainerT1 dst, size_t size)
+      -> decltype(q_interface.copy_to_host(src, dst, size)) {
+    return q_interface.copy_to_host(src, dst, size);
   }
-  /*  @brief Copying the data back to device
-      @tparam T is the type of the data
-      @param src is the device pointer we want to copy from.
-      @param dst is the host pointer we want to copy to.
-      @param size is the number of elements to be copied
+
+  /*  @brief Getting range accessor from the container
+      @tparam ContainerT0 is the type of the  data
+      @tparam AcM is the access mode
+      @param container is the  data we want to get range accessor
   */
-  template <typename T>
-  inline void copy_to_host(T *src, T *dst, size_t size) {
-    q_interface.copy_to_host(src, dst, size);
+  template <typename ContainerT0,
+            cl::sycl::access::mode AcM = cl::sycl::access::mode::read_write>
+  auto get_range_access(ContainerT0 container)
+      -> decltype(q_interface.template get_range_access<AcM>(container)) {
+    return q_interface.template get_range_access<AcM>(container);
   }
 
-  /*  @brief Copying the data back to device
-      @tparam T is the type of the data
-      @param src is the buffer_iterator we want to copy from.
-      @param dst is the host pointer we want to copy to.
-      @param size is the number of elements to be copied
+  /*  @brief waiting for a particular event
+      @param event is an instance of sycl::sycl::event
   */
-  template <typename T>
-  inline void copy_to_host(buffer_iterator<T> src, T *dst, size_t = 0) {
-    sycl_queue().submit([&](cl::sycl::handler &cgh) {
-      auto acc =
-          blas::get_range_accessor<cl::sycl::access::mode::read>(src, cgh);
-      cgh.copy(acc, dst);
-    });
+  template <typename first_event_t, typename... next_event_t>
+  inline void sync(first_event_t first_event, next_event_t... next_events) {
+    q_interface.wait_for_events(first_event, next_events...);
   }
 
-  template <typename T,
-            cl::sycl::access::mode AcM = cl::sycl::access::mode::read_write>
-  ContainerT<T, AcM> get_range_access(T *vptr) {
-    return q_interface.template get_range_access<T, AcM>(vptr);
-  }
-  template <typename T,
-            cl::sycl::access::mode AcM = cl::sycl::access::mode::read_write>
-  ContainerT<T, AcM> get_range_access(buffer_iterator<T> buff) {
-    return blas::get_range_accessor<AcM>(buff);
-  }
+  /*  @brief waiting for a sycl::queue.wait()
+   */
+  void inline sync() { q_interface.wait(); }
+
   /*!
    * @brief Executes the tree without defining required shared memory.
    */
   template <typename Tree>
   inline cl::sycl::event execute(Tree t) {
-    const auto localSize = 128;
+    const auto localSize = policy_handler().get_work_group_size();
```

Comment:
policy_handler should be "get_policy_handler" to avoid confusion with constructors

Reply:
fixed